detect_secrets_server/repos/base_tracked_repo.py

```diff
@@ -142,19 +142,24 @@ def scan(self, exclude_files_regex=None, exclude_lines_regex=None):
             exclude_lines=exclude_lines_regex,
         )
 
+        scan_from_this_commit = self.storage.get_empty_repo_commit_hash() if scan_head else self.last_commit_hash
         try:
-            diff = self.storage.get_diff(self.last_commit_hash)
+            diff_name_only = self.storage.get_diff_name_only(scan_from_this_commit)
+
+            # do a per-file diff + scan so we don't get a OOM if the the commit-diff is too large
+            for filename in diff_name_only:
+                file_diff = self.storage.get_diff(scan_from_this_commit, [filename])
```

Comment:
Might make sense to just pass the 1 filename (e.g. `file=filename`) and then do `[filename]` in the function, since this seems to be the only call that we pass files via.

Reply:
++